config/src/config/network_config.rs

```diff
@@ -54,23 +48,21 @@ pub struct NetworkConfig {
     #[serde(skip)]
     pub seed_peers: SeedPeersConfig,
     pub seed_peers_file: PathBuf,
+    // network_keypairs contains the node's network keypairs.
+    pub network_keypairs: Option<NetworkKeyPairs>,
```

Comment:
Note: if I leave this defined where it was previously (i.e., higher up in the list), some tests fail with "Unable to save config: values must be emitted before tables"; this seems to be an issue with TOML not liking tables (i.e., nested types), defined higher than values (e.g. https://docs.rs/toml/0.4.6/src/toml/ser.rs.html). It seems somewhat related to this (https://github.com/alexcrichton/toml-rs/issues/142), but I'm not exactly sure how to overcome this without it looking hacky...

Reply:
it may require some serde attributes on NetworkKeyPairs to work correctly. It might actually be better to split the keypairs and just take them directly. There's no value in having them bundled together.

Reply:
As discussed offline, there's no easy way to improve this right now. I've noted it down as something to address in the future, but let's avoid the issue by leaving network_keypairs at the bottom of the struct. I don't want this issue to block me entirely.